internal/nodes/collector/cv1.go

```diff
@@ -33,7 +33,7 @@ func (nc *Collector) summary(parentStreamTags []string, parentMeasurementTags []
 	}
 
 	req := clientset.CoreV1().RESTClient().Get().RequestURI(nc.baseURI + "/proxy/stats/summary")
-	res := req.Do()
+	res := req.Do(nc.ctx)
```

Comment:
This is my only concern with this. Maybe something to resolve in another PR though, since you did not create this issue. But, it is something we might want to refactor at some point. Including a context in a struct is usually an anti-pattern. And, in this case, the context is getting set on the Collector type by the Collect function, and then reused by all the internal function calls following. This would cause problems if Collector values ever needed to be used in a concurrent way. Obviously this is not the case now, since this works. But, at some point, it might be a good idea to refactor this code so that the functions in question all accept the context as the first parameter, and pass it as needed. This would be more idiomatic and resolve potential concurrent use issues, or potential issues of the context value getting accidentally used in the wrong way.

Reply:
I created https://circonus.atlassian.net/browse/CIRC-8952 to address this, thanks!